packages/notifier/src/types.js

```diff
@@ -61,20 +69,31 @@
  */
 
 /**
- * @typedef {any} NotifierInternals Purposely opaque. Will be shared between
- * machines, so it must be safe to expose. But other software should avoid
- * depending on its internal structure.
+ * @template {string} S
+ * @template P
+ * @typedef {{ _name: S, _param: P }} OpaqueParam An opaque type that conveys the
+ * type parameter P.
```

Comment:
Wow, this is a lot more mechanism than I expected. Use your judgement about whether it is worth it. If not, revert as many of my suggested changes as needed to restore simplicity.

Reply:
> Wow, this is a lot more mechanism than I expected.

Yeah, TypeScript is not built to hide types from folks, it really wants to expose as many as possible.

For simplicity, I'll go back to transparent sharable internals.  But I will just use `head` and `tail` directly, not `_head` and `_tail` since that's a breaking change.